File: vite-ecom-project/src/app/models/Basket.ts
import type { Product } from "./Product"

export type Basket = {
  id: string
  items: Item[]
}

export class Item {
  constructor(product: Product, quantity: number) {
    this.id = product.productId
    this.productId = product.productId
    this.quantity = quantity
    this.price = product.price
    this.name = product.name
    this.image = product.image
    this.description = product.description
    this.brand = product.brand
  }

  id: number
  productId: number
  quantity: number
  price: number
  name: string
  image: string
  description: string
  brand: string
}

File: api-ecom-project/api-ecom-project/Models/basket.cs
namespace ApiEcomProject.Models
{
    public class Basket
    {
        public Guid Id { get; set; }
        public List<BasketItem> Items { get; set; } = new List<BasketItem>();
    }
    public class BasketItem
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
        public string Brand { get; set; }
        public string Type { get; set; }
        public int Stock { get; set; }
    }
}
